kolibri/plugins/coach/assets/src/composables/useExerciseResources.js

```diff
@@ -0,0 +1,213 @@
+
+  function topicInChildrenHasMore() {}
+
+  function fetchMore() {
+    if (get(_moreParams)) {
+      fetchContentList(_moreParams);
+    } else {
+      console.error("No more params to fetch. There shouldn't be a 'Show More' button visible.");
+    }
+  }
+
+  /**
+   * @param {Boolean} appendResults Whether to append the results to the current list or replace it
+   */
+  function fetchContentList(moreParams = null) {
+    set(_loading, true);
+
+    // Set the default params for the fetchTree method .
+    const params = {
+      ...(moreParams || {}),
+      ...{
+        kind_in: get(_kindsToFetch),
+        include_coach_content: true,
+      },
+    };
+
+    ContentNodeResource.fetchTree({
+      id: get(_currentTopicId),
+      params,
+    }).then(topicTree => {
+      // results is the list of all children
+      // more is an object that contains the parameters we need to fetch the next batch of nodes
+      const { results, more } = topicTree.children;
+      const topicResults = results.filter(({ kind }) => kind === ContentNodeKinds.TOPIC);
+      const exerciseResults = results.filter(({ kind }) => kind === ContentNodeKinds.EXERCISE);
+
+      console.log(results);
+      console.log(more);
+
+      console.log('fetching descendant assessments');
+      ContentNodeResource.fetchDescendantsAssessments(results.map(topic => topic.id))
+        .then(({ data: topicsHasDescendantCounts }) => {
+          // Just get the IDs of the topics that have descendant exercises
+          const topicIdsWithDescendantCounts = topicsHasDescendantCounts
+            .filter(topic => topic.num_assessments > 0)
+            .map(topic => topic.id);
+
+          // Now, filter the topicResults to only include those topics
+          const topicsToAnnotate = topicResults.filter(topic =>
+            topicIdsWithDescendantCounts.includes(topic.id)
+          );
+
+          // Return those topics annotated to include the `num_assessments` property
+          return topicsToAnnotate.map(topic => {
+            topic.num_assessments = topicsHasDescendantCounts.find(
+              t => t.id === topic.id
+            ).num_assessments;
+            return topic;
+          });
+        })
+        .then(topicsWithCounts => {
+          const currentTopicContentList = topicsWithCounts.concat(exerciseResults);
+          if (moreParams) {
+            set(_contentList, [...get(_contentList), ...currentTopicContentList]);
+          } else {
+            set(_contentList, currentTopicContentList);
+          }
+        })
+        .then(() => set(_loading, false))
+        .catch(e => {
+          // TODO Work out best UX for this situation -- it may depend on if we're fetching more
+          // or the initial list of contents
+          console.error(e);
+          set(_loading, false);
+        });
+    });
+  }
+
+  /**
+   * @param {string} topicId The topicId for which we want to fetch the descendants
+   */
+  function setCurrentTopicId(topicId) {
```

Comment:
I think the logic here could be handled by directly watching the computed topicId I suggested above.